go/libraries/doltcore/sqle/enginetest/commit_amend_test.go

```diff
@@ -0,0 +1,189 @@
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+func TestCommitMessageAmend(t *testing.T) {
+	harness := newDoltHarness(t)
+	enginetest.TestScript(t, harness, queries.ScriptTest{
+		Name: "Sysbench Transactions Shouldn't Cause Constraint Violations",
+		SetUpScript: []string{
+			"SET @@AUTOCOMMIT=0;",
+			"CREATE TABLE test (id INT PRIMARY KEY );",
+			"INSERT INTO test (id) VALUES (1)",
+			"CALL DOLT_ADD('.');",
+			"CALL DOLT_COMMIT('-m', 'original commit message');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query: "SELECT  message FROM dolt_log;",
+				Expected: []sql.Row{
+					{"original commit message"},
+					{"checkpoint enginetest database mydb"},
+					{"Initialize data repository"},
+				},
+			},
+			{
+				Query:            "CALL DOLT_COMMIT('--amend', '-m', 'modified commit message');",
+				SkipResultsCheck: true,
+			},
+			{
+				Query: "SELECT  message FROM dolt_log;",
+				Expected: []sql.Row{
+					{"modified commit message"},
+					{"checkpoint enginetest database mydb"},
+					{"Initialize data repository"},
+				},
+			},
+		},
+	})
+}
+func TestCommitChangesAmend(t *testing.T) {
+	harness := newDoltHarness(t)
+	enginetest.TestScript(t, harness, queries.ScriptTest{
+		Name: "Sysbench Transactions Shouldn't Cause Constraint Violations",
+		SetUpScript: []string{
+			"SET @@AUTOCOMMIT=0;",
+			"CREATE TABLE test (id INT PRIMARY KEY );",
+			"INSERT INTO test (id) VALUES (1)",
+			"CALL DOLT_ADD('.');",
+			"CALL DOLT_COMMIT('-m', 'original commit message');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "SELECT COUNT(*) FROM dolt_status;",
+				Expected: []sql.Row{{0}},
+			},
+			{
+				Query: "SELECT  message FROM dolt_log;",
+				Expected: []sql.Row{
+					{"original commit message"},
+					{"checkpoint enginetest database mydb"},
+					{"Initialize data repository"},
+				},
+			},
+			{
+				Query:            "INSERT INTO test (id) VALUES (2)",
+				SkipResultsCheck: true,
+			},
+			{
+				Query:    "SELECT COUNT(*) FROM dolt_status;",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:            "CALL DOLT_ADD('.');",
+				SkipResultsCheck: true,
+			},
+			{
+				Query:            "CALL DOLT_COMMIT('--amend');",
+				SkipResultsCheck: true,
+			},
+			{
+				Query: "SELECT message FROM dolt_log;",
+				Expected: []sql.Row{
+					{"original commit message"},
+					{"checkpoint enginetest database mydb"},
+					{"Initialize data repository"},
+				},
+			},
+			{
+				Query:            "INSERT INTO test (id) VALUES (3)",
+				SkipResultsCheck: true,
+			},
+			{
+				Query:    "SELECT COUNT(*) FROM dolt_status;",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:            "CALL DOLT_ADD('.');",
+				SkipResultsCheck: true,
+			},
+			{
+				Query:            "CALL DOLT_COMMIT('--amend', '-m', 'updated commit message');",
+				SkipResultsCheck: true,
+			},
+			{
+				Query:    "SELECT COUNT(*) FROM dolt_status;",
+				Expected: []sql.Row{{0}},
+			},
```

Comment:
This is checking that the table changes aren't staged anymore, which is good, but it would be better to test that the commit itself contains the data change you added. I think the easiest way to do that would be to look into the `dolt_diff_test` system table, which will show you the exact data changes made in each commit. 

This is a pretty cool system table to know about. Here's some docs on it: https://docs.dolthub.com/sql-reference/version-control/dolt-system-tables#dolt_diff_usdtablename

Maybe something like `select to_id, from_id from dolt_diff_test limit 1` and then assert that `to_id` is 3 and `from_id` is NULL.

Reply:
Excellent advice, adding dolt_diff_BLAH checks for these tests.